onnx/__init__.py

```diff
@@ -62,12 +62,15 @@ def _serialize(proto: Union[bytes, google.protobuf.message.Message]) -> bytes:
     '''
     if isinstance(proto, bytes):
         return proto
-    elif hasattr(proto, 'SerializeToString') and callable(proto.SerializeToString):
-        result = proto.SerializeToString()
+    if hasattr(proto, 'SerializeToString') and callable(proto.SerializeToString):
+        try:
+            result = proto.SerializeToString()
+        except ValueError as e:
+            if proto.ByteSize() >= onnx.checker.MAXIMUM_PROTOBUF:
+                raise ValueError("The single proto is larger than 2GB. Please use save_as_external_data to save proto separately.") from e
+            raise
         return result
```

Comment:
@justinchuby I found we have different mypy version between required check (from style.sh) and Python Lint. That's why this line only passed by style.sh (mypy 0.782) but failed with Python Lint (mypy 0.971). To prevent confusion, is it possible to use mypy 0.782 from reviewdog for now? I believe many of existing ONNX code cannot pass by the latest mypy.

If it is not feasible, do you think we should disable no-any-return in Python Lint's mypy? It seems to complain here due to missing type specified from Protobuf's SerializeToString, which will be represented as Any type.

Reply:
It doesn't seem the check supports this at the moment. disabling no-any-return sounds like a good idea.

Reply:
You could try changing `result = proto.SerializeToString()' to `return proto.SerializeToString()` and delete the `return result` to see if it helps.

Reply:
Or, assign an empty string to `result` at the beginning. I wonder if it is just an imprecision in the analysis producing a false positive.

Reply:
If you really want, I would do `cast(bytes, result` to explicitly make mypy happy. https://docs.python.org/3/library/typing.html#typing.cast

Reply:
Mypy doesn't know about SerializeToString. It may if we compile the lib first and install all dependencies in the github action

Reply:
Thank you both for all the suggestions, but unfortunately none of these can resolve the mypy error in two CIs. `cast(bytes, result)` will bring Redundant cast to "bytes" if mypy knows SerializeToString. 

> Mypy doesn't know about SerializeToString. It may if we compile the lib first and install all dependencies in the github action

Good idea. I think we should do that before mypy check to solve this issue globally. Especially I also saw mypy errors in other PRs like:
```
[mypy] reported by reviewdog 🐶 Library stubs not installed for "google.protobuf.message" (or incompatible with Python 3.8) [import] Raw Output: /home/runner/work/onnx/onnx/onnx/helper.py:8:1: error: Library stubs not installed for "google.protobuf.message" (or incompatible with Python 3.8) [import]
```
and
```
[mypy] reported by reviewdog 🐶 Module "onnx" has no attribute "OptionalProto"; maybe "Optional"? [attr-defined] Raw Output: /home/runner/work/onnx/onnx/onnx/mapping.py:5:1: error: Module "onnx" has no attribute "OptionalProto"; maybe "Optional"? [attr-defined]
```

@justinchuby do you have bandwidth to take care of this in added mypy check from Python Lint? Definitely it can help to resolve many unneccessary mypy errors in every PR. Thanks!

Reply:
I may be able to pick this up early next week if that works.